assets/eip-7543/decimal_float_test.go

```diff
@@ -0,0 +1,346 @@
+	"fmt"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+)
+
+var PRECISION = uint256.NewInt(10)
+
+func (d *Decimal) String() string {
+	c := new(uint256.Int).Set(&d.c)
+	q := new(uint256.Int).Set(&d.q)
+	cs := ""
+	if c.Sign() == -1 {
+		cs = "-"
+		c.Neg(c)
+	}
+	qs := ""
+	if q.Sign() == -1 {
+		qs = "-"
+		q.Neg(q)
+	}
+	return fmt.Sprintf("%v%v*10^%v%v", cs, c.Dec(), qs, q.Dec())
+}
+
+func BenchmarkOpAdd(b *testing.B) {
+	intArgs := []*uint256.Int{uint256.NewInt(987349875), uint256.NewInt(987349875), uint256.NewInt(987349875), uint256.NewInt(987349875)}
+	benchmarkOpDec(b, intArgs, opAdd)
+}
+
+func BenchmarkOpDecAdd(b *testing.B) {
+	intArgs := []*uint256.Int{PRECISION, uint256.NewInt(987349875), uint256.NewInt(987349875), uint256.NewInt(987349875), uint256.NewInt(987349875)}
+	benchmarkOpDec(b, intArgs, opDecAdd)
+}
+
+func BenchmarkOpDecNeg(b *testing.B) {
+	intArgs := []*uint256.Int{uint256.NewInt(987349875), uint256.NewInt(987349875)}
+	benchmarkOpDec(b, intArgs, opDecNeg)
+}
+
+func BenchmarkOpDecMul(b *testing.B) {
+	intArgs := []*uint256.Int{PRECISION, uint256.NewInt(987349875), uint256.NewInt(987349875), uint256.NewInt(987349875), uint256.NewInt(987349875)}
+	benchmarkOpDec(b, intArgs, opDecMul)
+}
+
+func BenchmarkOpDecInv(b *testing.B) {
+	// opDecInv benchmark does not depend on precision
+	intArgs := []*uint256.Int{PRECISION, MINUS_ONE_INT256, uint256.NewInt(1)}
+	benchmarkOpDec(b, intArgs, opDecInv)
+}
+
+func BenchmarkOpDecExp(b *testing.B) {
+	// opDecExp benchmark depends on steps
+	steps := uint256.NewInt(10)
+	intArgs := []*uint256.Int{steps, PRECISION, uint256.NewInt(0), uint256.NewInt(1)}
+	fmt.Println("BenchmarkOpDecExp steps=", steps)
+	benchmarkOpDec(b, intArgs, opDecExp)
+}
+
+func BenchmarkOpDecLn(b *testing.B) {
+	// opDecExp benchmark depends on steps
+	steps := uint256.NewInt(10)
+	intArgs := []*uint256.Int{steps, PRECISION, uint256.NewInt(0), uint256.NewInt(2)}
+	fmt.Println("BenchmarkOpDecLn steps=", steps)
+	benchmarkOpDec(b, intArgs, opDecLn)
+}
+
+func BenchmarkOpDecSin(b *testing.B) {
+	// opDecExp benchmark depends on precision
+	steps := uint256.NewInt(10)
+	intArgs := []*uint256.Int{steps, PRECISION, uint256.NewInt(0), uint256.NewInt(1)}
+	fmt.Println("BenchmarkOpDecSin steps=", steps)
+	benchmarkOpDec(b, intArgs, opDecSin)
+}
+
+func benchmarkOpDec(b *testing.B, intArgs []*uint256.Int, op executionFunc) {
+	var (
+		env            = NewEVM(BlockContext{}, TxContext{}, nil, params.TestChainConfig, Config{})
+		stack          = newstack()
+		scope          = &ScopeContext{nil, stack, nil}
+		evmInterpreter = NewEVMInterpreter(env)
+	)
+
+	env.interpreter = evmInterpreter
+
+	pc := uint64(0)
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		for _, arg := range intArgs {
+			stack.push(arg)
+		}
+		op(&pc, evmInterpreter, scope)
+		stack.pop()
+		stack.pop()
+	}
+	b.StopTimer()
+}
+
+func TestSignedCmp(t *testing.T) {
+	var gas uint64
+
+	// b := uint256.NewInt(15)
```

Comment:
pls cleanup the commented code which is not relevant

Reply:
removed